pkg/controller/import-controller.go

```diff
@@ -181,6 +183,7 @@ func NewImportController(mgr manager.Manager, log logr.Logger, importerImage, pu
 		verbose:         verbose,
 		pullPolicy:      pullPolicy,
 		recorder:        mgr.GetEventRecorderFor("import-controller"),
+		cdiNamespace:    util.GetNamespace(),
```

Comment:
I would rename `client` in this function to `c` or something like that (to avoid name collision with the package), and add a namespacedClient to the reconciler.
Then you can do something like this:
```go
reconciler.namespacedClient = client.NewNamespacedClient(c, r.cdiNamespace)
```
And in the function that gets the config map in the CDI namespace, you can now use the namespacedClient and it will work. You will have a cached namespaced client instead of an uncached version.

Reply:
It looks like `namespacedClient.Get()` simply [calls](https://github.com/kubevirt/containerized-data-importer/blob/main/vendor/sigs.k8s.io/controller-runtime/pkg/client/namespaced_client.go#L141-L153) the original `client.Get()`,  so it doesn't do any better than the original code, and we get the same good old errors:
```
I1212 22:35:17.495054       1 reflector.go:255] Listing and watching *v1.ConfigMap from pkg/cache/internal/informers_map.go:250
W1212 22:35:17.501040       1 reflector.go:324] pkg/cache/internal/informers_map.go:250: failed to list *v1.ConfigMap: configmaps is forbidden: User "system:serviceaccount:cdi:cdi-sa" cannot list resource "configmaps" in API group "" at the cluster scope
E1212 22:35:17.501184       1 reflector.go:138] pkg/cache/internal/informers_map.go:250: Failed to watch *v1.ConfigMap: failed to list *v1.ConfigMap: configmaps is forbidden: User "system:serviceaccount:cdi:cdi-sa" cannot list resource "configmaps" in API group "" at the cluster scope
```
I changed to k8s client and it seems to fix the issue.